graderx/graders/manager.py

```diff
@@ -214,74 +256,108 @@ def delete_course(course_id):
     del courses_config[course_id]
     update_course_config(courses_config)
 
+
 def delete_lab(course_id, lab_id):
     courses_config = get_courses_config_if_course_exists(course_id)
     try:
         stdout_common.delete_lab(course_id, lab_id)
     except FileNotFoundError:
         pass
-    courses_config[course_id]["labs"] = list(filter(lambda lab: lab['name'] != lab_id, courses_config[course_id]["labs"]))
+    courses_config[course_id][COURSE_LABS] = list(
+        filter(lambda lab: lab[LAB_NAME] != lab_id, courses_config[course_id][COURSE_LABS]))
     update_course_config(courses_config)
 
+
 def sanitize_and_validate_lab_data(lab_data):
-    if 'name' not in lab_data or len(lab_data['name']) > 20:
+    if LAB_NAME not in lab_data or len(lab_data[LAB_NAME]) > 20:
```

Comment:
same note for `COURSE_NAME`, this should be `LAB_NAME_KEY` since it holds the same value as the current `COURSE_NAME` you can just name both as `NAME_KEY`.